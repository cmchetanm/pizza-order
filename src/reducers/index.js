import { combineReducers } from "redux";
import DoughTypes from "./reducer_dough_types";
import activeDoughType from './reducer_active_dough';
import Ingredients from './reducer_ingredient_list';
import ActiveIngredients from './reducer_active_ingredient'

const rootReducer = combineReducers({
  doughTypes: DoughTypes,
  activeDoughTypes: activeDoughType,
  ingredients: Ingredients,
  activeIngredients: ActiveIngredients
});

export default rootReducer;
