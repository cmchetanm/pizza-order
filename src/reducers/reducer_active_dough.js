const initialAction = {
  type: null
}

export default function(state = null, action = initialAction) {
  switch (action.type) {
    case "DOUGH_SELECTED":
      return action.payload;
    default:
      return state
  }
}
