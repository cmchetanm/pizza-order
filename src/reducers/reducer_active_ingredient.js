// State argument is not application state, only the state
// this reducer is responsible for
const initialState = {
  totalPrice: null,
  ingredients: []
}

const initialAction = {
  type: null
}

function removeByKey(array, params){
  array.some(function(item, index) {
    return (array[index].id === params.id) ? !!(array.splice(index, 1)) : false;
  });
  return array;
}

function addIngredient(state= null, action) {
  state.ingredients.push(action.payload)
  state.price = state.ingredients.reduce((prev, next) => prev + next.price, 0);
  return {totalPrice: state.price, ingredients: state.ingredients};
}

function removeIngredient(state= null, action) {
  state.ingredients = removeByKey(state.ingredients, action.payload)
  state.price = state.ingredients.reduce((prev, next) => prev + next.price, 0);
  return {totalPrice: state.price, ingredients: state.ingredients};
}

export default function(state = initialState, action = initialAction) {
  switch (action.type) {
    case "INGREDIENT_SELECTED":
      return addIngredient(state, action);
    case 'INGREDIENT_REMOVED':
      return removeIngredient(state, action);
    default:
      return state
  }
}
