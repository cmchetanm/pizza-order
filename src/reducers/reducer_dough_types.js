export default function() {
  return [
    {
      id: 1,
      name: 'Thin',
      description: 'Thin and crusty pizza dough',
      price: 0.5

    },
    {
      id: 2,
      name: 'Thick',
      description: 'Thick and crusty pizza dough',
      price: 1
    }
  ];
}
