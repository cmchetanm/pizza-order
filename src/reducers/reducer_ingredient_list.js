export default function() {
  return [
    {
      id: 1,
      name: 'pepperoni',
      displayName: 'Pepperoni',
      price: 1
    },
    {
      id: 2,
      name: 'anchovies',
      displayName: 'Anchovies',
      price: 10
    },
    {
      id: 3,
      name: 'lobster',
      displayName: 'Lobstah',
      price: 25
    },
    {
      id: 4,
      name: 'truffle oil',
      displayName: 'truffle oil',
      price: 100
    }
  ];
}
