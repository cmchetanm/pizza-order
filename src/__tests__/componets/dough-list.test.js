import React from 'react'
import { shallow, mount, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import DoughList from "../../containers/dough-list";
import { createStore } from "redux";
import reducers from "../../reducers";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const store = reducers();

test('Renders broker code', () => {
  const renderedValue =  renderer.create(<DoughList store={createStore(reducers)} />).toJSON()
  expect(renderedValue).toMatchSnapshot();
});
