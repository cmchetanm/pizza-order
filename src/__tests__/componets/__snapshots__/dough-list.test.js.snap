// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders broker code 1`] = `
<div
  className="row"
>
  <div
    className="list-group-item col-sm-6 dough-names"
    onClick={[Function]}
  >
    Thin
  </div>
  <div
    className="list-group-item col-sm-6 dough-names"
    onClick={[Function]}
  >
    Thick
  </div>
</div>
`;
